Automattic/wp-calypso
# Focused Launch: Add Launch your site button

### Changes proposed in this Pull Request

- Display _"Launch your site"_ button sticky and full-width on mobile
- Disable _"Launch your site"_ button if at least one of the active steps doesn't have a valid selected value. Enable it otherwise.
- When pressing the button, show the loading screen first (success screen without buttons and with slightly different copy) cc: @ollierozdarz since this is not in designs; added to Screenshots section below.
- When the site is launched, show the Success view.
- If the user selected paid items during Launch flow, show Checkout overlay.
- Ignore any previous items existing in cart (eg: additional domains) and populate the cart only with selected items from Launch flow.

#### Todo (or follow up):
- [x]  When redirecting the user back to the editor from the thank you page or they navigate themselves to editor from plans page / My home / other pages in Calypso, Success View should be displayed unless the user clicks one of the action buttons to close it. (https://github.com/Automattic/wp-calypso/pull/47808)
- [x] Enable _"Launch your site"_ button if there is a paid plan and selected domain or paid plan and domain. https://github.com/Automattic/wp-calypso/pull/47601#discussion_r529772106
- [x] When pressing _"Launch your site"_ button, don't show checkout if:
   - there is a purchased plan and a purchased domain
   - there is a purchased plan and subdomain selected

## Testing instructions
1. Check out the branch on your machine and run in two parallel terminal windows:
  - `yarn && yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
2. Add an _unlaunched_ site created with `/start` to your sandbox (for convenience, `SITE_ID`)
3. Visit `calypso.localhost:3000/page/SITE_ID/home?flags=create/focused-launch-flow` and click on the "Launch" button. The Focused Launch modal should appear.

#### Testing instructions (Focused launch free and paid flows)
4. Select free domain and plan and click "Launch your site" at the bottom => the loading view and then Success view should appear.
5. Repeat steps 2-3 and select a paid domain and a paid plan => the loading view and then checkout overlay should be displayed. Closing the checkout overlay should reveal the Success view so you can continue editing (no redirect).
6. Repeat steps 2, 3, 5 and complete purchase for a plan. You should be redirected to thank-you page (depending on the plan) and after skipping the upsell you should be redirected back to editor (WIP to show success view here).

#### Testing instructions (regression testing - Step by step launch)
8. Create a site on a free plan starting at `/new` and sandbox it.
9. Press launch in editor and launch flow with sidebar should appear.
10. Complete the flow by selecting a free plan => redirect to My home
11. Complete the flow by selecting a paid plan => redirect to `/checkout` with some known error mentioned in #47150

* Note: testing this in calypso overlay (using `?flags=create/focused-launch-flow-calypso` flag) will launch the site, add any selected paid items to cart, but won't open Checkout modal.

### Screenshot
<img width="702" alt="Screenshot 2020-11-24 at 11 50 05" src="https://user-images.githubusercontent.com/14192054/100077682-51849380-2e4b-11eb-8e53-cceb417dc25b.png">

Fixes #47392
Fixes #47241
Fixes #47734
